cfme/tests/openstack/cloud/test_instances.py

```diff
@@ -0,0 +1,176 @@
+"""Tests for Openstack cloud instances"""
+
+import fauxfactory
+import pytest
+import random
+
+from cfme.cloud.instance.openstack import OpenStackInstance
+from cfme.cloud.provider.openstack import OpenStackProvider
+from cfme.web_ui import Quadicon
+from utils import testgen
+from utils.appliance.implementations.ui import navigate_to
+from utils.version import current_version
+
+pytest_generate_tests = testgen.generate([OpenStackProvider],
+                                         scope='module')
+
+pytestmark = [pytest.mark.usefixtures("setup_provider_modscope")]
+
+
+@pytest.mark.usefixtures("setup_provider_modscope")
+@pytest.fixture(scope='function')
+def active_instance(provider):
```

Comment:
Sorry but I don't agree with this fixture at all. Its taking an already running instance, or picking a random one and trying to start it, and then the tests are shutting the instance down (regardless of its initial state) and never recovering the initial state.  What if it picks an instance in use by a test running in parallel? Or a C&U instance that should be on 24x7?

Drop this and provision an instance to test on every time, or use a module instance and change the tests to get the instance in the starting state.